src/reducer.js

```diff
@@ -14,22 +14,17 @@ function createHistory (state, ignoreInitialState) {
   return history
 }
 
-// lengthWithoutFuture: get length of history
-function lengthWithoutFuture (history) {
-  return history.past.length + 1
-}
-
 // insert: insert `state` into history, which means adding the current state
 //         into `past`, setting the new `state` as `present` and erasing
 //         the `future`.
 function insert (history, state, limit, group) {
   debug.log('inserting', state)
-  debug.log('new free: ', limit - lengthWithoutFuture(history))
+  debug.log('new free: ', limit - history.past.length - 1)
```

Comment:
I think now it is a bit unclear why we are subtracting one. Either we store this in a separate variable (`history.path.length + 1`), or we keep the function